libs/render_delegate/light.cpp

```diff
@@ -526,10 +531,10 @@ void HdArnoldGenericLight::Sync(HdSceneDelegate* sceneDelegate, HdRenderParam* r
                 param->Interrupt();
                 interrupted = true;
                 const AtString oldName{AiNodeGetName(_light)};
-                AiNodeDestroy(_light); // TODO should we also remove it from the delegate list ???  !!!!
+                if (!_delegate->GetProceduralParent())
```

Comment:
I think it would be cool to have a function `_delegate->DestroyNode(node)` ,  a bit like `_delegate->CreateNode()`. That would allow to avoid repeating the pattern `if (!_delegate->GetProceduralParent()) {AiDestroy();}` and allow to add more logic later on in one place if needed.